packages/tools/accessibility/src/accessibilityGUIItem.ts

```diff
@@ -0,0 +1,113 @@
+import { Vector2 } from "core/Maths/math";
+import { Button } from "gui/2D/controls/button";
+import { Control } from "gui/2D/controls/control";
+import { Image } from "gui/2D/controls/image";
+import { TextBlock } from "gui/2D/controls/textBlock";
+import { Vector2WithInfo } from "gui/2D/math2D";
+import { AccessibilityItem } from "./accessibilityItem";
+
+/**
+ * A abstract layer to store the accessibility tree structure. It is constructed from the BabylonJS scene entities that need to be accessible. It informs the parent-children relationship of accessibility tree, and informs how to render: description, isActionable, onclick/onrightclick/onfocus/onblur.
+ */
+export class AccessibilityGUIItem extends AccessibilityItem {
+    /**
+     * The corresponding BabylonJS entity. Can be a Node or a Control.
+     */
+    public entity: Control;
+
+    /**
+     * The children of this item in the accessibility tree.
+     */
+    public children: AccessibilityGUIItem[];
+
+    constructor(entity: Control, children: AccessibilityGUIItem[]) {
+        super(entity, children);
+    }
+
+    /**
+     * The text content displayed in HTML element.
+     */
+    public override get description(): string {
+        let description = "";
+        if (this.entity.accessibilityTag?.description) {
+            description = this.entity.accessibilityTag.description;
+        }
+        else if (this.entity instanceof TextBlock) {
+            description = (this.entity as TextBlock).text;
+        }
+        else if (this.entity instanceof Button) {
+            description = (this.entity as Button).textBlock?.text ?? "";
+        }
+        else if (this.entity instanceof Image) {
+            description = (this.entity as Image).alt ?? "";
+        }
+        return description;
+    }
+
+    /**
+     * If this entity is actionable (can be clicked).
+     */
+    public override get isActionable(): boolean {
+        if (this._isActionable) {
+            return this._isActionable;
+        }
+
+        if (this.entity instanceof Button) {

```

Comment:
Could this be set in the class constructor instead of the get?

Reply:
It's doing a lazy instanciation.